sled-agent/src/sim/instance.rs

```diff
@@ -47,33 +84,113 @@ impl Simulatable for SimInstance {
 
     fn request_transition(
         &mut self,
-        target: &InstanceRuntimeStateRequested,
+        target: &InstanceStateRequested,
     ) -> Result<Option<InstanceAction>, Error> {
-        self.state.request_transition(target)
-    }
-
-    fn execute_desired_transition(&mut self) -> Option<InstanceAction> {
-        if matches!(self.state.current().run_state, InstanceState::Rebooting) {
-            self.state.observe_transition(&PropolisInstanceState::Starting)
-        } else if let Some(desired) = self.state.desired() {
-            // These operations would typically be triggered via responses from
-            // Propolis, but for a simulated sled agent, this does not exist.
-            //
-            // Instead, we make transitions to new states based entirely on the
-            // value of "desired".
-            let observed = match desired.run_state {
-                InstanceStateRequested::Running => {
-                    PropolisInstanceState::Running
+        match target {
+            InstanceStateRequested::Running => match self.terminal_state() {
```

Comment:
Hmm, could we change `terminal_state`'s return type to be more explicit about what's the actual terminal state we're currently headed to vs the current state. Otherwise reading this match almost makes it seem like `Creating`/`Starting`/etc are valid terminal state when they aren't.

Reply:
Oh, though I suppose this is using terminal to mean "last" here rather than a steady state.

Reply:
I renamed this to "next resting state" in 892f0b56 to try to clarify this.